internal/rules/main_test.go

```diff
@@ -0,0 +1,64 @@
+package rules_test
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/joho/godotenv"
+	"github.com/namsral/flag"
+	"github.com/titpetric/factory"
+
+	systemMigrate "github.com/crusttech/crust/system/db"
+)
+
+func TestMain(m *testing.M) {
+	// @todo this is a very optimistic initialization, make it more robust
+	godotenv.Load("../../.env")
+
+	prefix := "system"
+	dsn := ""
+
+	p := func(s string) string {
+		return prefix + "-" + s
+	}
+
+	flag.StringVar(&dsn, p("db-dsn"), "crust:crust@tcp(db1:3306)/crust?collation=utf8mb4_general_ci", "DSN for database connection")
+	flag.Parse()
+
+	if testing.Short() {
+		return
+	}
+
+	factory.Database.Add("default", dsn)
+
+	db := factory.Database.MustGet()
+	db.Profiler = &factory.Database.ProfilerStdout
+
+	// migrate database schema
+	if err := systemMigrate.Migrate(db); err != nil {
+		log.Printf("Error running migrations: %+v\n", err)
+		return
+	}
+
+	os.Exit(m.Run())
+}
+
+func assert(t *testing.T, ok bool, format string, args ...interface{}) bool {
```

Comment:
I've started with abstracting this `assert` code repetition into `internal/test.Assert`

Reply:
Noted, TODO for a cleanup sprint. A few suggestions:

1. rename ErrNil into NoError, - previously `must()`
2. add Error(), - previously `mustFail()`

I looked at [testify](https://godoc.org/github.com/stretchr/testify/require) for the naming here.

Reply:
If we can find something better with a smaller footprint sure. Maybe I'm being too pedantic, but 3 functions vs. a package with ~100 of them... I'll leave it to you, since you started with this, so +1 if you want ;)